### PR TITLE
feature: new header option to emit the headers in fromQuery

### LINKED ISSUE #2752: [Enhancement] Add an option to `fromQuery` to emit the column headers as well
This feature request stems from an internal discussion for capturing the header names in the SQL query results.

### Ideal behavior 

For a sample CSV file-based data source 

```
cat <<EOF > test.csv
foo,bar
1,hello
2,ciao
3,hola
4,bonjour
EOF

```

The following snippet should emit the header 

```nextflow
Channel
    .sql
    .fromQuery("SELECT * FROM CSVREAD('test.csv') where foo>=2;", header: true) // <---- OPTIONAL header 
    .collectFile(name: 'sample.txt', newLine: true) { row -> row.join(',') }


```

Which would yield the output 

```csv
foo,bar        <---- HEADER
3,hola
2,ciao
4,bonjour


```

### PR DESCRIPTION
**Pull request by jorgeaguileraseqera**
resolve #2752

Signed-off-by: Jorge Aguilera <jorge.aguilera@seqera.io>

### CODE REVIEW

**Comment by pditommaso** [comment]
This is good. Just renamed `header` option to `emitColumns`, or `emitColumnName` could be even better. @jorgeaguileraseqera @abhi18av what do you think?

**Comment by abhi18av** [comment]
Looks good to me Paolo 💯

**Comment by jorgeaguileraseqera** [comment]
@pditommaso you need to sign your commit

**Comment by pditommaso** [comment]
You are right!

File: modules/nextflow/src/main/resources/nextflow/trace/assets/moment.min.js [hunk 1]
```diff
@@ -1 +1,2 @@
...
...,t){return fn(e,t,"monthsShort")},f.weekdaysMin=function(e,t,n){return mn(e,t,n,"weekdaysMin")},f.defineLocale=ft,f.updateLocale=function(e,t){var n,s;return null!=t?(s=ot,null!=R[e]&&null!=R[e].parentLocale?R[e].set(X(R[e]._config,t)):(t=X(s=null!=(n=dt(e))?n._config:s,t),null==n&&(t.abbr=e),(s=new K(t)).parentLocale=R[e],R[e]=s),ct(e)):null!=R[e]&&(null!=R[e].parentLocale?(R[e]=R[e].parentLocale,e===ct()&&ct(e)):null!=R[e]&&delete R[e]),R[e]},f.locales=function(){return ee(R)},f.weekdaysShort=function(e,t,n){return mn(e,t,n,"weekdaysShort")},f.normalizeUnits=_,f.relativeTimeRounding=function(e){return void 0===e?Yn:"function"==typeof e&&(Yn=e,!0)},f.relativeTimeThreshold=function(e,t){return void 0!==On[e]&&(void 0===t?On[e]:(On[e]=t,"s"===e&&(On.ss=t-1),!0))},f.calendarFormat=function(e,t){return(e=e.diff(t,"days",!0))<-6?"sameElse":e<-1?"lastWeek":e<0?"lastDay":e<1?"sameDay":e<2?"nextDay":e<7?"nextWeek":"sameElse"},f.prototype=i,f.HTML5_FMT={DATETIME_LOCAL:"YYYY-MM-DDTHH:mm",DATETIME_LOCAL_SECONDS:"YYYY-MM-DDTHH:mm:ss",DATETIME_LOCAL_MS:"YYYY-MM-DDTHH:mm:ss.SSS",DATE:"YYYY-MM-DD",TIME:"HH:mm",TIME_SECONDS:"HH:mm:ss",TIME_MS:"HH:mm:ss.SSS",WEEK:"GGGG-[W]WW",MONTH:"YYYY-MM"},f});
```
[the remark this answers is not in the document]
Comment (pditommaso) [reply modules/nextflow/src/main/resources/nextflow/trace/assets/moment.min.js:1]: ignore

**Comment by pditommaso** [comment]
I've merged this manually 6a6a6eac because I had some conflicts with another pending feature I was working, see 3a9dad8b. Thanks a lot!

**Pull request closed without merging**